### DISCUSSION TITLE
🎚️ Multiplayer Room Filter: Difficulty cap, mandatory and banned MODs, etc.

### DISCUSSION THREAD

**Discussion by Bryceed** [ppy/osu#16790] [Ideas]
Just as the "host rotate" was implemented natively in the game, there could be global filters in the room, ensuring that everyone respects the rules of the room to add songs:

- Possibility of query parameter(s) should be always applied when a player tries to add a song in queue (like limiting between X~Y stars, AR>7, BPM=<200, etc.);
- Forcefully enable and ban mods for selection in the room, regardless of the individual selection of each user (can be any mod or all);
- Lock and unlock gamerule change in room (like standard to mania)

Just like happens in stable, it's annoying when someone enters the room and doesn't even respect what is explicit in the title of the room, which is annoying both for those who created the room and for those who are there playing, hoping that those rules are respected (_and what's more still annoying, having to keep talking about it all the time, again and again_).

This would benefit all levels of players, since in the same way that there will be no high level entering the beginners room to trolling them to play like "The Big Black" or "Centipede", there will also not be beginners who will put on poorly mapped songs or lower star rating (_leaving the room when the beatmap starts, because he doesn't know how to play it yet, trolling too who was in the room to play difficult songs_).

*_Having a mod filter would not impact extra mods as it would only affect mods that are mandatory and banned for all matches in the room (while the filter is active). It would still be possible to force a Hard Rock and ban mods from difficulty settings for all songs in the queue, and it would still be fine for the user to choose to play with Flashlight or Autopilot mod, if extra mods are active._

**Comment by peppy** [comment]
Definitely a direction I think we want to go in.
